src/app/socket/participantSocket/participant.socket.ts

```diff
@@ -0,0 +1,35 @@
+import {DefaultSocket} from "../defaultSocket/default.socket";
+import {Question} from "../../model/question/question.model";
+import {Injectable} from "@angular/core";
+import {Observable} from "rxjs";
+import {Participant} from "../../model/participant/participant.model";
+
+@Injectable({
+  providedIn: 'root'
+})
+export class ParticipantSocket extends DefaultSocket {
+
+  async connect(token: string) {
+    try {
+      const frame = await super.connect(token);
+      return Promise.resolve(frame);
+    }catch (e){
+      return Promise.reject(e);
+    }
+  }
+
+  addQuestion(question: Question){
+    this.getStompClient().send("/admin/question/add", {}, question.text);
+  }
+
+  public onQuestion(): Observable<Question>
+  {
+    return this.subscribe<Question>('/admin/question')
+  }
+
+  public onJoinParticipant(sessionId: number): Observable<Participant>
+  {
+    return this.subscribe<Participant>('/participant/session/'+sessionId+'/user/onjoin');
```

Comment:
Kann hier ein Fehler fliegen, falls der return nicht in ein Observable<Participant> umgewandelt werden kann? 
Wenn ja wo wird dieses aufgefangen?

Reply:
Wird mit dem Ticket LFD-38 upgedatet.